src/test/java/org/opendatakit/common/persistence/QueryResultTest.java

```diff
@@ -361,10 +363,21 @@ public void testCase3() throws ODKDatastoreException {
         int fetchSizes[] = {1, 2, 1021, 303, 101, 2831 };
         int idxFetch = 4;
         int len = 0;
-        QueryResult result = query.executeQuery(null, fetchSizes[idxFetch]);
+        // It seems that the following two asserts try to test if the results
+        // are marks as having more rows on the database than the ones retrieved
+        // by the query
+        // This was the original query:
+        // QueryResult result = query.executeQuery(null, fetchSizes[idxFetch]);
+        // This is the modified query ensuring that the page size is inferior to
+        // the number of rows on the database
+        QueryResult result = query.executeQuery(null, 10);
         len += result.getResultList().size();
         assertEquals( true, result.hasMoreResults());
-        assertEquals( fetchSizes[idxFetch], result.getResultList().size());
+        // This was the original assertion, which dependened on some fixed
+        // values having sense in relation to what's inserted as test fixtures

```

Comment:
I don't understand this part. `values having sense in relation to what's inserted as test fixtures`